src/axom/primal/geometry/Vector.hpp

```diff
@@ -516,6 +539,38 @@ AXOM_HOST_DEVICE inline void Vector<T, NDIMS>::negate()
   }
 }
 
+template <typename T, int NDIMS>
+AXOM_HOST_DEVICE inline void Vector<T, NDIMS>::normalize()
+{
+  const double len_sq = squared_norm();
+
+  if(len_sq >= primal::PRIMAL_TINY)
+  {
+    m_components /= (std::sqrt(len_sq));
+  }
+  else
+  {
+    // TODO: Would it be better to do nothing if the vector is really the
+    //       zero vector? Or if it is tiny but not the zero vector, find
+    //       the component with the largest magnitude and set it to 1 (all
+    //       the other components would then be set to 0).
+
+    // Set the first component to 1 and all others to 0
+    m_components[0] = static_cast<T>(1.);
+
+    for(int i = 1; i < NDIMS; ++i)
+    {
+      m_components[i] = static_cast<T>(0.);
+    }
+  }
+}
```

Comment:
Should `Vector::unitVector()` be implemented in terms of this one?

Here's the current implementation:
https://github.com/LLNL/axom/blob/379db95ceed3ce8e90145993a0d9c33f8103f330/src/axom/primal/geometry/Vector.hpp#L489-L507